### PR TITLE
add chat

### PR DESCRIPTION
**Pull request by kcinnaJlol**

### CODE REVIEW

**Review by watchful5406 — CHANGES_REQUESTED** [review]
chat should close when the user clicks anywhere else but it

**Review by watchful5406 — CHANGES_REQUESTED** [review]
new chat messages should disappear after some time to avoid clogging up the screen (or make it possible to close)

**Comment by SakuruWasTaken** [comment]
> chat should close when the user clicks anywhere else but it



> new chat messages should disappear after some time to avoid clogging up the screen (or make it possible to close)


these should be moved over to https://github.com/OpenLWR/OpenLWR/pull/11

File: server/client_events/client_chat_event.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import json
+from server.server_events import server_chat_event
+
+def handle(data, source_user):
+    server_chat_event.fire("{1}: {0}".format(data, source_user))
```
Comment (SakuruWasTaken) [inline server/client_events/client_chat_event.py:5]: i don't think it's a great idea to just send the message over without doing any kind of validation at all, if i wanted to annoy users by sending a message 30000 chars long, there'd be nothing to stop me from doing that here.

(i'll add this on my own after merge, you don't need to do anything here)

**Review by SakuruWasTaken — APPROVED** [review]

**Review by watchful5406 — APPROVED** [review]

**Pull request merged**